include/StringTools.h

```diff
@@ -0,0 +1,294 @@
+    String MEZZ_LIB CamelCaseCopy(String Source);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Size Modifying String Manipulations
+
+    /// @brief Trims all whitespaces and tabs from a one or both sides of a String.
+    /// @param Source The String to be modified.
+    /// @param Left Whether or not to trim the left side of the String.
+    /// @param Right Whether or not to trim the right side of the String.
+    void MEZZ_LIB Trim(String& Source, const Boole Left = true, const Boole Right = true);
+    /// @brief Trims all whitespaces and tabs from a one or both sides of a String.
+    /// @param Source The String that will be the basis for the modified String.
+    /// @param Left Whether or not to trim the left side of the String.
+    /// @param Right Whether or not to trim the right side of the String.
+    String MEZZ_LIB TrimCopy(String Source, const Boole Left = true, const Boole Right = true);
+    /// @brief Replaces all instances of multiple consecutive whitespaces with only a single whitespace.
+    /// @param Source The String to be modified.
+    void MEZZ_LIB RemoveDuplicateWhitespaces(String& Source);
+    /// @brief Gets a copy of a string with all duplicate whitespaces removed.
+    /// @param Source The String that will be the basis for the modified String.
+    String MEZZ_LIB RemoveDuplicateWhitespacesCopy(String Source);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Colour Channels
+
+    /// @brief Converts a string containing hex to a ColourValue channel.
+    /// @param Hex The Hex value to be converted. Must be a 2 character String.
+    /// @return Returns a Real representing the converted Hex string.
+    Real ConvertHexToColourChannel(const StringView Hex);
+    /// @brief Converts a ColourValue channel to Hex.
+    /// @remarks This method will produce a two character String of hexidecimal values. If needed, the "0x" prefix
+    /// would have to be applied manually.
+    /// @param Channel The value to be converted to Hex.
+    /// @return Returns a two character string containing the hex expression for the provided channel value.
+    String ConvertColourChannelToHex(const Real Channel);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Tokenizing
+
+    /// @brief Splits a string into multiple substrings based on the specified delimiters.
+    /// @param Source The String to be split.
+    /// @param Delims The characters to look for and use as split points in the source String.
+    /// @param MaxSplits The maximum number of splits to perform on this String. Value of zero means unlimited splits.
+    /// @return Returns a vector containing all the substrings generated from the source String.
+    StringVector MEZZ_LIB Split(const String& Source, const String& Delims = " \t\n", const Whole MaxSplits = 0);
```

Comment:
Do we have a matching: String Join(StringVector ToJoin, const String& Delim=" ");

Js, Ruby, Python, and PHP all have some function to match their split/explode functions that join groups into a single string following a simple pattern of ToJoin[0] + Delim +ToJoin[1] + Delim ... ToJoin[size-1];

Reply:
We do now!